test/buy-tshirt.spec.ts

```diff
@@ -1,48 +1,68 @@
 import { browser } from 'protractor';
 // tslint:disable-next-line: max-line-length
 import { MenuContentPage, ProductListPage, ProductAddedModalPage, SummaryStepPage, SignInPage, AddressPage, ShippingPage, PaymentPage, BankPaymentPage, OrderSummaryPage } from '../src/page';
+describe('Buying a T-shirt', () => {
 
-describe('Buy a t-shirt', () => {
-  const email = 'j.l-h@hotmail.com';
-  const password = 'vkvkvkvk';
-  const menuContentPage: MenuContentPage = new MenuContentPage();
-  const productListPage: ProductListPage = new ProductListPage();
-  const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
-  const summaryStepPage: SummaryStepPage = new SummaryStepPage();
-  const signInPage: SignInPage = new SignInPage();
-  const addressPage: AddressPage = new AddressPage();
-  const shippingPage: ShippingPage = new ShippingPage();
-  const paymentPage: PaymentPage = new PaymentPage();
-  const bankpaymentPage: BankPaymentPage = new BankPaymentPage();
-  const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
+  describe('Opening web page', () => {
+    it('Web page loads correctly', async() => {
```

Comment:
use it only when you have asserts, if you need to execute some commands into a describe user `beforeEach` or `beforeAll`

Reply:
Aren't DOM actions (clicking, writing info on the UI, etc.) kind of an assert by themselves? I used this structure because when the click commands are part of the core function i'm testing and when they are inside an it() and any throws an error, you can clearly see where the problem is on the report but when using only one it(), you don't exactly know where it is.
Isn't it an indirect assert i can put inside an it()?

Reply:
un it always has at least an assert, in your case you are splitting the test into several describe to have more readability, and your "its" doesn't have assert

if you have it without assert, you are breaking "AAA" standard

Reply:
kind of an assert by themselves? no, the assert should be explicit... `Assert. equals` for example, also the DOM action does not fail controlled... a good assert should have a clear message for example `Expected 5 but was 3`